install.hs

```diff
@@ -241,87 +280,84 @@ stackInstallHie versionNumber = do
 
 buildCopyCompilerTool :: VersionNumber -> Action ()
 buildCopyCompilerTool versionNumber =
-  execStackWithYaml_ versionNumber ["build", "--copy-compiler-tool"]
+  execStackWithGhc_ versionNumber ["build", "--copy-compiler-tool"]
 
 stackTest :: VersionNumber -> Action ()
-stackTest versionNumber = execStackWithYaml_ versionNumber ["test"]
+stackTest versionNumber = execStackWithGhc_ versionNumber ["test"]
 
 stackBuildDoc :: Action ()
-stackBuildDoc = generateHoogleDatabase $ do
-  execStack_ ["--stack-yaml=shake.yaml", "install", "hoogle"]
-  execStack_ ["--stack-yaml=shake.yaml", "exec", "hoogle", "generate"]
+stackBuildDoc = do
+  execStackShake_ ["build", "hoogle"]
+  execStackShake_ ["exec", "hoogle", "generate"]
 
+-- | short help message is printed by default
 shortHelpMessage :: Action ()
 shortHelpMessage = do
+  hieVersions <- getHieVersions
   let out = liftIO . putStrLn
   scriptName <- liftIO getProgName
   out ""
   out "Usage:"
   out' ("stack " <> scriptName <> " <target>")
   out ""
   out "Targets:"
-  mapM_ (out' . showTarget spaces) targets
+  mapM_ (out' . showTarget (spaces hieVersions)) (targets hieVersions)
```

Comment:
This should show the nightly ghc version as well. That's my last nitpick, afterwards we can merge.